app/src/main/java/com/example/triptracker/view/map/AddSpot.kt

```diff
@@ -0,0 +1,443 @@
+import com.example.triptracker.model.location.Pin
+import com.example.triptracker.navigation.compareDistance
+import com.example.triptracker.view.theme.Montserrat
+import com.example.triptracker.view.theme.md_theme_dark_error
+import com.example.triptracker.view.theme.md_theme_dark_gray
+import com.example.triptracker.view.theme.md_theme_light_black
+import com.example.triptracker.view.theme.md_theme_light_onPrimary
+import com.example.triptracker.view.theme.md_theme_orange
+import com.example.triptracker.viewmodel.RecordViewModel
+import com.google.android.gms.maps.model.LatLng
+
+@Composable
+/**
+ * AddSpot is a composable function that allows the user to add a new spot to the path. This will
+ * only be shown when pressing + in the record screen
+ *
+ * @param recordViewModel: RecordViewModel
+ * @param latLng: LatLng at where the spot is going to be added
+ */
+fun AddSpot(recordViewModel: RecordViewModel, latLng: LatLng) {
+
+  // Variables to store the state of the add spot box
+  var boxDisplayed by remember { mutableStateOf(true) }
+
+  // Variables to store the state of the location string of the spot
+  var location by remember { mutableStateOf("") }
+
+  // Variables to store the state of the description of the spot
+  var description by remember { mutableStateOf("") }
+
+  // Variables to store the state of the position of the spot in form of latlng
+  var position by remember { mutableStateOf(latLng) }
+
+  // Variable to store the state of the error
+  var isError by remember { mutableStateOf(false) }
+
+  // Variables to store the state of the selected pictures
+  var selectedPictures by remember { mutableStateOf<List<Uri?>>(emptyList()) }
+
+  // Launcher for the pick multiple media activity
+  val pickMultipleMedia =
+      rememberLauncherForActivityResult(ActivityResultContracts.PickMultipleVisualMedia(5)) {
+          pictures ->
+        // Callback is invoked after the user selects media items or closes the
+        // photo picker.
+        if (pictures.isNotEmpty()) {
+          Log.d("PhotoPicker", "Number of items selected: ${pictures.size}")
+          selectedPictures = pictures
+        } else {
+          Log.d("PhotoPicker", "No media selected")
+        }
+      }
+
+  // Get the point of interest of the current location if it exists
+  LaunchedEffect(Unit) { recordViewModel.getPOI(latLng) }
+
+  when (boxDisplayed) {
+    true ->
+        Box(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(15.dp)
+                    .background(color = md_theme_light_black, shape = RoundedCornerShape(35.dp))
+                    .testTag("AddSpotScreen")) {
+              Column(modifier = Modifier.matchParentSize()) {
+
+                // Close button
+                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.End) {
+                  IconButton(
+                      modifier = Modifier.padding(10.dp), onClick = { boxDisplayed = false }) {
```

Comment:
Since this is a pop up that will show only on the top of the record map view it is ok to open it like this. Moreover we don't want to quit this screen when the "back" button of the phone is pressed since this would result in a loss of information and the latlng would be updated. I think this is ok for now :)